src/lib/arch/aarch64/Instruction_execute.cc

```diff
@@ -199,8 +204,7 @@ void Instruction::execute() {
     case Opcode::AArch64_ADDv1i64: {  // add dd, dn, dm
       const uint64_t n = operands[0].get<uint64_t>();
       const uint64_t m = operands[1].get<uint64_t>();
-      uint64_t out[2] = {static_cast<uint64_t>(n + m), 0};
-      results[0] = out;
+      results[0] = {n + m, 256};
```

Comment:
`256` in the second statement is the `bytes` parameter in the `RegisterValue` constructor. I am using the [list initialization syntax.](https://en.cppreference.com/w/cpp/language/list_initialization) I chose this form in the interest of verbosity, but I realize that this could be confusing.

`result[0] = out` is equivalent to ` result[0] = RegisterValue(out)`. The constructor is called implicitly in the first case. The only change is that the size is passed as well.

See explanation below as to why an array is used and why the second element is 0.

Reply:
OK this makes sense, thanks for clarifying